companion/gateway/agents/appraisal.py

```
@@ -0,0 +1,465 @@
+                    arousal=response_json.get('arousal', 0.0),
+                    dominance=response_json.get('dominance', 0.0)
+                )
+            else:
+                # If JSON parsing fails, fall back to basic rule-based system
+                return self.calculate_emotion_delta(event, {})
+        except asyncio.TimeoutError:
+            self.logger.warning("LLM appraisal timed out, falling back to rule-based")
+            return self.calculate_emotion_delta(event, {})
+        except Exception as e:
+            # Log the full traceback for debugging
+            self.logger.exception(f"AI-enhanced appraisal failed: {e}. Falling back to basic appraisal.")
```

Ruff (0.13.2)

Redundant exception object included in `logging.exception` call

(TRY401)